### PR TITLE
Convert size-0 reduction into FullOp

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/arith.cpp [hunk 1]
```diff
@@ -1225,16 +1225,58 @@ TensorView* reductionOpNoSqueeze(
...
+TensorView* maybeFullInsteadOfReduction(
+    const std::vector<unsigned int>& axes, // sorted
+    Val* init,
+    TensorView* tv,
+    bool keep_dim,
+    DataType dtype) {
+  auto tv_root = TensorDomain::noReductions(tv->getMaybeRFactorDomain());
+  const int ndims = tv_root.size();
+  for (auto i : axes) {
+    if (tv_root[i]->extent()->isZeroInt()) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/arith.cpp]: nit: use `at` for extra safety

File: torch/csrc/jit/codegen/cuda/arith.cpp [hunk 2]
```diff
@@ -1274,6 +1316,12 @@ TensorView* reductionOp(
   }
   std::sort(uint_axes.begin(), uint_axes.end());
 
+  auto maybe_full =
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/arith.cpp:1321]: Please leave a comment about this behavior, i.e., in PyTorch size-0 reduction effectively means a full op.

**Review by naoyam — APPROVED** [review]
LGTM

File: torch/csrc/jit/codegen/cuda/arith.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/arith.cpp]: changed

File: torch/csrc/jit/codegen/cuda/arith.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/arith.cpp:1321]: added

**Pull request merged**